entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -89,14 +97,54 @@ public void execute(EntityQueryRequest request, StreamObserver<ResultSetChunk> r
         .convertToEDSQuery(request, attrNameToEDSAttrMap.get(request.getEntityType()));
     Iterator<Document> documentIterator = entitiesCollection.search(
         DocStoreConverter.transform(tenantId.get(), query));
-    List<Entity> entities = convertDocsToEntities(documentIterator);
-
-    //Build result
-    //TODO : chunk response. For now sending everything in one chunk
-    responseObserver.onNext(convertEntitiesToResultSetChunk(
-        entities,
-        request.getSelectionList(),
-        attrNameToEDSAttrMap.get(request.getEntityType())));
+
+    if (!documentIterator.hasNext()) {
+      ResultSetChunk.Builder resultBuilder = ResultSetChunk.newBuilder();
+      resultBuilder.setResultSetMetadata(ResultSetMetadata.newBuilder()
```

Comment:
Can we build the ResultSetMetadata once (it shouldn't change between chunks or cases) and avoid special casing the no data case? In case it's helpful, here's the QS implementation of chunking: https://github.com/hypertrace/query-service/blob/main/query-service-impl/src/main/java/org/hypertrace/core/query/service/RowChunkingOperator.java

Reply:
Building ResultSetMetadata once. 

I think the initial block is fine, I can get away with by moving to a `do while loop`, but that would be unnecessary.